### PR TITLE
Add Flux OCI documentation and example workflows

### PR DESCRIPTION
**Pull request by stefanprodan**
Preview: https://deploy-preview-1073--fluxcd.netlify.app/docs/cheatsheets/oci-artifacts/

Part of: https://github.com/fluxcd/flux2/issues/2948

### CODE REVIEW

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+### Helm OCI
+
+For Helm users, Flux comes with [support](/docs/guides/helmreleases/#helm-oci-repository) for
+defining Helm releases with charts stored in container registries.
+
+Unlike Helm, the Flux OCI artifacts don't have a custom media type and they can be stored in
+any container registry. The artifacts created by Flux can contain any type of configuration
+besides Kubernetes manifests.
+
+The two artifacts types can be used together, for example you could bundle a `Namespace`,
```
Comment (darkowlzz) [inline content/en/docs/cheatsheets/oci-artifacts.md:62]:
```suggestion
The two artifact types can be used together, for example you could bundle a `Namespace`,
```
I also wonder if it'd be better to say "source artifact", more explicit.

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+a `HelmRepository` and a `HelmRelease` into a Flux OCI artifact where the `HelmRepository`
+points to a container registry where the chart is.
+
+## Workflow examples
+
+What follows are examples for deploy a demo application to Kubernetes using manifests stored in Git.
```
Comment (darkowlzz) [inline content/en/docs/cheatsheets/oci-artifacts.md:68]: maybe
```suggestion
Following are examples for deploying a demo application to Kubernetes using manifests stored in Git.
```

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+```mermaid
+flowchart LR
+
+A((me)) --> B(Git Repository)
```
Comment (darkowlzz) [inline content/en/docs/cheatsheets/oci-artifacts.md:73]: maybe
```suggestion
A((User)) --> B(Git Repository)
```
or even Developer?

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+specify the Git source and revision with: 
+
+```shell
+flux push artifact oci://<repo url> --path=<manifests dir> \
+	--source="$(git config --get remote.origin.url)" \
+	--revision="$(git branch --show-current)/$(git rev-parse HEAD)"
+```
+
+The Git source and the revision are stored in the container registry as annotations in the OCI artifact manifest,
+and they are reflected in-cluster under the `status.artifact.metadata`.
```
Comment (darkowlzz) [inline content/en/docs/cheatsheets/oci-artifacts.md:286]: maybe
```suggestion
and they are reflected in-cluster under the `status.artifact.metadata` of the `OCIRepository` object.
```
just to make it more clear.

**Comment by darkowlzz** [comment]
Since the cheatsheet is about OCI in general and also mentions about Helm OCI, should we put an example of how to use Helm OCI as well? Not sure what example may be useful. I may be wrong but I believe that the Helm OCI related examples are snippets in the HelmRepository source spec docs and don't have a complete example. But at the same time, I do think there's not much to show, just set the proper URL and type.
Maybe the recently added support for specifying just the registry name in HelmRepository and path to chart in HelmChart/HelmRelease (refer https://github.com/fluxcd/source-controller/pull/856) could be documented here.

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+E --> F[Staging]
+E --> G[Production]
+```
+
+We'll use the main branch to deploy to staging and Git tags for deploying on production.
+When pushing changes to the main branch, a CI job generates the Kubernetes manifests for staging and
+pushes them to a container registry using the FLux CLI. Flux running on the staging cluster
+detects the new artifact digest, pulls the manifests and applies them.
+
+When cutting a release from them main branch by Git tagging a commit with a semver version, the CI job
```
Comment (somtochiama) [inline content/en/docs/cheatsheets/oci-artifacts.md:86]:
```suggestion
When cutting a release from the main branch by tagging a commit with a semver version, the CI job
```

**Comment by stefanprodan** [comment]
> should we put an example of how to use Helm OCI as well?

@darkowlzz that's a great suggestion, we lack a simple example for this. I've added it, please check it out.

File: content/en/docs/cheatsheets/oci-artifacts.md [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+      chart: podinfo
+      version: 6.1.0 # {"$imagepolicy": "flux-system:podinfo-chart:tag"}
+      sourceRef:
+        kind: HelmRepository
+        name: podinfo
+```
+
+Based on the above configuration, Flux will scan the container registry every five minutes,
+and when it finds a newer Helm chart version, it will update the
+`HelmRelease.spec.chart.spec.chart.version`and will push the change to Git.
```
Comment (pjbgf) [inline content/en/docs/cheatsheets/oci-artifacts.md:537]:
```suggestion
`HelmRelease.spec.chart.spec.chart.version` and will push the change to Git.
```

**Review by pjbgf — COMMENTED** [review]
Just a few tiny nits, apart from that LGTM.

**Review by darkowlzz — APPROVED** [review]
LGTM!

**Review by pjbgf — APPROVED** [review]
LGTM

Great job @stefanprodan! :clap:

**Pull request merged**